### DISCUSSION TITLE
Newbie struggling to run a test

### DISCUSSION THREAD

**Discussion by GilesDMiddleton** [dotnet/runtime#51443] [Q&A] [answered]
Hi - I've just spent about 5 hours and finally managed to build the entire repo, build the libs in release, and build the tests.

But I've been trying to follow the instructions on how to run tests in visual studio, but no tests are showing in the Visual Studio Test Explorer. - What solution file should I be loading? I tried loading an individual test .csproj file, but it wouldn't compile as it's missing dependencies.

A little stumped!

Is there a simple walkthrough?

**Comment by SingleAccretion** [comment, marked as answer]
Have you tried the `./build -vs <Project name>` method?

**Reply by GilesDMiddleton** [reply]
Hey, 

Thanks for replying - that worked!

I didn't see that syntax mentioned in any of the initial .md files, all they said is that I needed to open the appropriate project and tests would appear.

My next challenge is to work out what existing tests there might be that cover #47463  - which I thought would be an easy first issue for me to get my wings, or at least practice working in this repo.

**Reply by SingleAccretion** [reply]
FWIW, the `-vs` switch is documented [here](https://github.com/dotnet/runtime/tree/main/docs/workflow/building/libraries#build), but if you know of some changes to some of the docs that would have made the process of finding that out more straightforward, I think that'd be a great contribution.

> My next challenge is to work out what existing tests there might be that cover #47463 here - which I thought would be an easy first issue for me to get my wings, or at least practice working in this repo.

Heh, that looks like a tricky path to test to me. I am not someone who could provide you with the guidance on this one.

**Reply by danmoseley** [reply]
I agree, if we were to make that change I think we would just verify it by eyeball.

You are welcome to have a go at that one, or consider looking at issues marked "easy":

https://github.com/dotnet/runtime/issues?q=is%3Aopen+is%3Aissue+label%3Aeasy

If you like you can tell us the kind of work you are looking for and maybe myself or @SingleAccretion  can help you find something that interests you. Welcome to our community.

**Reply by GilesDMiddleton** [reply]
Thanks for your help.

I also encountered a build issue once the project had loaded - and found that the .NET 4.8 targets was missing. After adding it I can now run the tests through VS2019 Test Explorer.
I see that is missing from the standard .vsconfig, and needs a "Microsoft.Net.Component.4.8.TargetingPack", 
When I was in the VSInstaller, I also ticked the python and CMake components, so maybe they are missing too and no need to install separately?

Would you like me to add the 4.8 pack as a PR? My first ever contribution!

@danmoseley The issue looked easy to me at first, simplest fix would be to check the pointer.. But I wanted to have a look at the typical pattern of usage and see if the class should be more robust, and possibly use any standard macros/asserts to validate programmers aren't calling methods when the m_rcEventThread hasn't been initialized. Quite a few of the methods don't check the pointer. As this my first exposure to the code base, I'm taking baby steps. I deal day-in-day out with C++/COM/SQL/C# for the last 20 years, I'm mostly interested in picking low hanging fruit and giving something back in the C++ realm to start with. 
nullptr checks are a great start.
